src/script/lua_api/l_internal.h

```diff
@@ -77,3 +79,12 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define GET_PLAIN_ENV_PTR         \
 	MAP_LOCK_REQUIRED;            \
 	GET_PLAIN_ENV_PTR_NO_MAP_LOCK
+
+// LuaJIT FFI function prototypes.
+#if USE_LUAJIT
+#define FFI_FCT(rettype, name, ...) \
+	extern "C" EXPORT_ATTRIBUTE rettype mtffi_##name(__VA_ARGS__) noexcept
+#else
+#define FFI_FCT(rettype, name, ...) \
+	UNUSED_ATTRIBUTE static rettype mtffi_##name(__VA_ARGS__) noexcept
+#endif
```

Comment:
Not relevant for the functions in this PR, but in general the function bodies need to be in try-catch blocks (we need lua errors on exceptions, not abort). A macro for this might be handy.